### PR TITLE
Feature: Add Gitalk Support

### PR DESCRIPTION
**Pull request by mirari**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/iissnan/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes have been added (for bug fixes / features).
- [ ] Docs have been added / updated (for bug fixes / features).

## PR Type
**What kind of change does this PR introduce?**  

- [ ] Bugfix.
- [x] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number(s): N/A

## What is the new behavior?
* Screens with this changes: N/A
* Link to demo site with this changes: 
[Demo](https://mirari.github.io/2017/08/14/%E5%88%97%E8%A1%A8%E5%85%83%E7%B4%A0%E6%97%A0%E5%86%85%E5%AE%B9%E6%97%B6%E9%AB%98%E5%BA%A6%E4%B8%BA0%E5%AF%BC%E8%87%B4%E4%B8%B2%E8%A1%8C%E7%9A%84%E9%97%AE%E9%A2%98/)

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

**Comment by ivan-nginx** [comment]
@mirari it seems, this new «version» of Gitment, right? Can u please give short description what difference `Gitalk` and `Gitment`? As i see, gitalk have cool interface, anything else?

**Comment by mirari** [comment]
@ivan-nginx They look the same but they are 'competitive products'.
 'Gitalk' supports multiple languages and multiple admins.
 I choose 'Gitalk' just for chinese UI.

**Comment by whtis** [comment]
哈哈，今天折腾完才看到有人pull了request，Gitalk的好处在于移动端也能正常使用，且有人在一直维护。gitment的作者貌似已经弃坑了。

附上我的开发笔记：[给hexo博客nexT主题添加Gitalk评论系统](http://www.whtis.com/2017/10/19/给hexo博客nexT主题添加Gitalk评论系统/)

**Comment by zjuturtle** [comment]
这个pull request 有点问题。layout/_partials/comments.swig 这个文件的修改

~~~
{% elseif theme.gitalk.enable %}
      <div id="gitalk-container"></div>
      <link rel="stylesheet" href="https://unpkg.com/gitalk/dist/gitalk.css">
    {% endif %}
  </div>
{% endif %}
~~~
会导致Gemini主题下，文章中的文章列表显示靠左对齐。


改成这样就不会了（虽然我也不太懂前端）

~~~
  {% elseif theme.gitalk.enable %}
    <div id="gitalk-container"></div>
    <link rel="stylesheet" href="https://unpkg.com/gitalk/dist/gitalk.css">
  {% endif %}
{% endif %}
~~~
@mirari

**Comment by xuchengpeng** [comment]
wonderful

**Comment by xuchengpeng** [comment]
这个加快进度赶紧合入哟 @mirari 赶紧跟跟催催哟

**Comment by AlynxZhou** [comment]
@mirari @ivan-nginx This pr seems good, could you please solve the conflicts so it can be merged?

**Comment by AlynxZhou** [comment]
@ivan-nginx If there are some conflicts and the author cannot fix it, could I create another PR to add Gitalk? I want to use this:-P

**Comment by ivan-nginx** [comment]
U can, of course. I just think it's will conflict with Gitmint adaptation. I not like gitment adaptation.

File: layout/_third-party/comments/gitalk.swig [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+          admin: '{{theme.gitalk.admin}}'.split(','),
+          pagerDirection: '{{theme.gitalk.pagerDirection}}',
+          // facebook-like distraction free mode
+          distractionFreeMode: false
+        })
+        gitalk.render('gitalk-container')
+      </script>
+    {% endif %}
+  {% endif %}
+{% endif %}
```
Comment (Raymoonkie) [inline layout/_third-party/comments/gitalk.swig]: {% if not (theme.duoshuo and theme.duoshuo.shortname) and not theme.duoshuo_shortname %}
  {% if theme.gitalk.enable %}
    {% if page.comments %}
      <script src="https://unpkg.com/gitalk/dist/gitalk.min.js"></script>
      <script src="https://cdn.bootcss.com/blueimp-md5/2.10.0/js/md5.js"></script> 
      <script type="text/javascript">
        const gitalk = new Gitalk({
          clientID: '{{theme.gitalk.clientID}}',
          clientSecret: '{{theme.gitalk.clientSecret}}',
          repo: '{{theme.gitalk.repo}}',
          owner: '{{theme.gitalk.owner}}',
          admin: '{{theme.gitalk.admin}}'.split(','),
          pagerDirection: '{{theme.gitalk.pagerDirection}}',
          id: md5(window.location.pathname),
          distractionFreeMode: false
        })
        gitalk.render('gitalk-container')
      </script>
    {% endif %}
  {% endif %}
{% endif %}

Comment (Raymoonkie) [reply layout/_third-party/comments/gitalk.swig]: 这里有一个小bug，github 新增了 对 label 的限制，不能超过50。按照当前配置，当文章题目稍长时，会报Error: Validation Failed。有关该问题的详细讨论参见[Gitalk issue 102](https://github.com/gitalk/gitalk/issues/102)，解决办法参见[calpa blog](https://calpa.me/2018/03/10/gitalk-error-validation-failed-442-solution/)。
主要新增了id 属性，对其进行md5 加密。简单解决了问题，至于要解决和之前的issue评论兼容的问题，上面的博客中也有解决办法。因为我不懂js，也就没做改动。

**Comment by ivan-nginx** [comment]
Already added in NexT v6.

***

NexT is rebased into [organization](https://github.com/theme-next) [repo](https://github.com/theme-next/hexo-theme-next).
If you want new feature, please, [create pull in NexT v6.x repo](https://github.com/theme-next/hexo-theme-next/pulls).
There is instructions on [English](https://github.com/theme-next/hexo-theme-next/blob/master/docs/UPDATE-FROM-5.1.X.md) or [Chinese](https://github.com/theme-next/hexo-theme-next/blob/master/docs/zh-CN/UPDATE-FROM-5.1.X.md) how to update from v5.1.x to v 6.x

You also may read [this](https://github.com/iissnan/hexo-theme-next/issues/2061) for details.

**Pull request closed without merging**